src/Chapter2.hs

```diff
@@ -727,7 +742,8 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate (x:xs) = replicate x x ++ smartReplicate xs
+smartReplicate [] = []
```

Comment:
You can notice that you need to get through each element of the list, turn it into the list of repeating elements and then concatenate all such lists together. This is exactly what you are doing 👍🏼 

I just want to mention, that you could use standard functions to do the same as you do as well: `map`.

However after using `map (\x -> replicate x x)` on the list you will get a list of lists. To flatten such a list, you can use the `concat` function:

```haskell
ghci> concat [[3, 1, 2], [4, 0, 1]]
[3,1,2,4,0,1]
```

This pattern of using `map` and `concat` is very common, so the standard library has even a standard function `concatMap` which is a combination of `map` and `concat` (as the name suggests).